core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -249,4 +268,9 @@ final public boolean hasTop() {
   public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
     throw new UnsupportedOperationException();
   }
+
+  @Override
+  public void close() throws Exception {
+    sources.forEach(s -> s.closeSafely());
+  }
```

Comment:
I think it's probably rare to have multiple sources, and this is really the only value of closeSafely. I think maybe we shouldn't have closeSafely at all, because it forces implementations to think implement it safely, and also they must decide when it's appropriate to use it when closing their sources.

Instead of having that, we should probably just do something special here in the case of multiple sources, to handle and aggregate exceptions from multiple sources.